nvidia_gpu_stats/cmd/nvidia_gpu_stats/main.go

```diff
@@ -0,0 +1,241 @@
+package main
+
+import (
+	"encoding/json"
+	"flag"
+	"fmt"
+	"os"
+	"os/signal"
+	"sync"
+	"syscall"
+	"time"
+
+	"github.com/shirou/gopsutil/v4/process"
+
+	"github.com/NVIDIA/go-nvml/pkg/nvml"
+)
+
+type GPUNvidia struct {
+	pid      int
+	nvmlInit nvml.Return
+}
+
+func NewGPUNvidia(pid int) *GPUNvidia {
+	g := &GPUNvidia{pid: pid}
+	return g
+}
+
+func (g *GPUNvidia) gpuInUseByProcess(device nvml.Device) bool {
```

Comment:
since this accepts a device, should it be named `gpuInUseByDevice`?

add doc explaining what it returns too